### DISCUSSION TITLE
Unexpected behaviour running a downloaded release after having built it locally

### DISCUSSION THREAD

**Discussion by janeklb** [gitx/gitx#337] [General]
I suspect that this might be standard behaviour when dealing with Xcode.. but I'm inexperienced with building xcode projects so I'm not sure. Could anyone confirm?

Here's what happened:
- downloaded latest release (0.22)
- moved it into `/Applications` (and replaced the older version)
- opened up a terminal session, and ran `gitx` from the command line (GitX opens)
- selected "About GitX" from the GitX menu to ensure it was running the downloaded version, but noticed that it was running an older version ❗ (`0.13-xxxxx` afair) 
- checked `which gitx`, which shows `/usr/local/bin/gitx`
- checked `ls -l $(which gitx)`, which shows that `/usr/local/bin/gitx` is a symlink to `/Applications/GitX.app/Contents/Resources/gitx`
- closed GitX
- opened GitX via Spotlight (cmd-space); "About GitX" now shows 0.22
- opened `gitx` via the cli; "About GitX" now shows the old version again
- `ps -ef | grep -i gitx` shows that I'm running `/Users/janeklb/Library/Developer/Xcode/DerivedData/GitX-aqlqktsdnvbaaafnhydtdvkmtcll/Build/Products/Debug/gitx`
  - i had built GitX earlier this year as part of https://github.com/gitx/gitx/issues/239 though that was my first time playing around with Xcode
- closed GitX and opened again via Spotlight
- `ps -ef | grep -i gitx` shows that I'm running `/Applications/GitX.app/Contents/Resources/gitx`

I then moved + archived `/Users/janeklb/Library/Developer/Xcode/DerivedData/GitX-aqlqktsdnvbaaafnhydtdvkmtcll` and tried running `gitx` from the CLI, and was presented with some errors (unfortunately I don't have the output - I had cleared the terminal). After I restarted my macbook - and possibly did some other things that I don't recall - and then encountered https://github.com/gitx/gitx/issues/336 whenever trying to run `gitx`. I've downgraded to 0.21 for now - this discussion post isn't about that issue.

So, it looks like I've been running a locally-built GitX since February, despite downloading new `GitX.app` releases and placing them in `/Applications`!

Is this standard behaviour when dealing with Xcode projects that you have built locally and download too?

**Comment by hannesa2** [comment]
Solved with https://github.com/gitx/gitx/releases/tag/0.23

**Reply by janeklb** [reply]
@hannesa2 can you please confirm whether my question is actually "solved" with 0.23, or are you solely referring to #336?

**Reply by hannesa2** [reply]
Did you tried `0.23` ? For me it works

**Reply by janeklb** [reply]
0.23 works fine - but that's not what my question is about. Don't worry if you don't have time to engage with it though, it's not strictly related to GitX - i was solely asking out of curiosity about some of behaviour i saw.

**Comment by hannesa2** [comment]
@janeklb Is this solved for you ?
I guess it's solved with https://github.com/gitx/gitx/pull/376

**Reply by janeklb** [reply]
I no longer have xcode set up on my local so am unable to test, but based on the issue you've linked it seems likely. Thanks!
